front/inbox.go

```diff
@@ -57,78 +57,80 @@ func dailyPosts(w text.Writer, r *request, day time.Time) {
+					)
 				)
-			)
-		left join (
-			select id from notes where author = $1
-		) myposts
-		on
-			myposts.id = notes.object->>'inReplyTo'
+			union
+			select replies.id, replies.object, replies.author, replies.inserted, replies.cc0, replies.to0, replies.cc1, replies.to1, replies.cc2, replies.to2, null as followed, persons.actor, null as g from
+			(select * from notes where author != $1 and inserted >= $4 and inserted < $4 + 60*60*24) replies
+			join
+			(select * from notes where author = $1 and inserted >= $4 and inserted < $4 + 60*60*24) mynotes
+			on
+				mynotes.id = replies.object->>'inReplyTo'
+			join
+			persons
+			on
+				persons.id = replies.author
+		) anotes
 		left join (
-			select notes.object->>'inReplyTo' as inreplyto, notes.author, follows.id as follow from notes left join follows on follows.follower = $1 and follows.followed = notes.author and follows.accepted = 1 where notes.inserted >= $3
+			select notes.object->>'inReplyTo' as inreplyto, notes.author, follows.id as follow from notes left join follows on follows.followed = notes.author where follows.follower = $1 and follows.accepted = 1 and notes.inserted >= $3
 		) replies
 		on
-			replies.inreplyto = notes.id and replies.author != notes.author
-		left join persons
+			replies.inreplyto = anotes.id and replies.author != anotes.author
+		left join
+		(
+			select author, max(inserted) / 86400 as last, count(*) / $2 as avg from notes where inserted >= $3 group by author
+		) stats
 		on
-			persons.id = notes.author
-		where
-			notes.inserted >= $4 and
-			notes.inserted < $4 + 60*60*24 and
-			(follows.followed is not null or myposts.id is not null)
-		group by notes.id
+			stats.author = anotes.author
+		group by anotes.id
 		order by
-			notes.inserted / 86400 desc,
+			anotes.inserted / 86400 desc,
 			(case
-				when notes.to0 = $1 and notes.to1 is null and notes.cc0 is null then 0
-				when $1 in (notes.cc0, notes.to0, notes.cc1, notes.to1, notes.cc2, notes.to2) or (notes.to2 is not null and exists (select 1 from json_each(notes.object->'to') where value = $1)) or (notes.cc2 is not null and exists (select 1 from json_each(notes.object->'cc') where value = $1)) then 1
+				when anotes.to0 = $1 and anotes.to1 is null and anotes.cc0 is null then 0
+				when $1 in (anotes.cc0, anotes.to0, anotes.cc1, anotes.to1, anotes.cc2, anotes.to2) or (anotes.to2 is not null and exists (select 1 from json_each(anotes.object->'to') where value = $1)) or (anotes.cc2 is not null and exists (select 1 from json_each(anotes.object->'cc') where value = $1)) then 1
 				else 2
 			end),
 			count(distinct replies.follow) desc,
 			count(distinct replies.author) desc,
-			follows.avg asc,
-			follows.last asc,
-			notes.inserted / 3600 desc,
-			persons.actor->>'type' = 'Person' desc,
-			notes.inserted desc
+			stats.avg asc,
+			stats.last asc,
+			anotes.inserted / 3600 desc,
+			anotes.actor->>'type' = 'Person' desc,
+			anotes.inserted desc
```

Comment:
[ ] Check why this query returns zero rows but only when running from Go - it works in sqlite shell